### PR TITLE
Say crossref resolved only when no `?`.

### LINKED ISSUE #68: SATySFi says that all cross references were solved while they aren't
Consider the following `mwe.saty`. Note that ``\ref(`sec:fooo`);`` is a typo on purpose.

```
@require: stdja

StdJa.document (|
  title = {};
  author = {};
  show-title = false;
  show-toc = false
|) '<
  +section ?:(`sec:foo`) {Foo} <
    +p {
      This is Section \ref(`sec:fooo`);.
    }
  >
>
```

If we typeset this document, we obtain the log shown below and it claims `all cross references were solved.` after the 2nd trial.

```console
$ satysfi mwe.saty -o mwe.pdf
 ---- ---- ---- ----
  target file: 'mwe.pdf'
  dump file: 'mwe.satysfi-aux' (will be created)
  parsing 'mwe.saty' ...
  parsing 'stdja.satyh' ...
  parsing 'pervasives.satyh' ...
  parsing 'gr.satyh' ...
  parsing 'list.satyh' ...
  parsing 'math.satyh' ...
  parsing 'color.satyh' ...
 ---- ---- ---- ----
  reading 'pervasives.satyh' ...
  type check passed.
 ---- ---- ---- ----
  reading 'list.satyh' ...
  type check passed.
 ---- ---- ---- ----
  reading 'color.satyh' ...
  type check passed.
 ---- ---- ---- ----
  reading 'math.satyh' ...
  type check passed.
 ---- ---- ---- ----
  reading 'gr.satyh' ...
  type check passed.
 ---- ---- ---- ----
  reading 'stdja.satyh' ...
  type check passed.
 ---- ---- ---- ----
  reading 'mwe.saty' ...
  type check: document
 ---- ---- ---- ----
  evaluating texts ...
  evaluation done.
 ---- ---- ---- ----
  breaking contents into pages ...
  needs another trial for solving cross references...
 ---- ---- ---- ----
  evaluating texts (2nd trial) ...
  evaluation done.
 ---- ---- ---- ----
  breaking contents into pages ...
  all cross references were solved.
 ---- ---- ---- ----
  embedding fonts ...
 ---- ---- ---- ----
  writing pages ...
 ---- ---- ---- ----
  output written on 'mwe.pdf'.
```

However, SATySFi obviously cannot find a label entry for `sec:fooo` and a question mark is printed on the output PDF. This situation wouldn't be considered as "solved" IMHO.

### PR DESCRIPTION
**Pull request by qnighy**
Fixes #68.

@gfngfn could you review this?

### CODE REVIEW

File: src/frontend/logging.ml [hunk 1]
```diff
@@ -67,11 +67,14 @@ let needs_another_trial () =
...
+  print_endline ("  could not reach to fixpoint when resolving cross references.")
 
 
-let achieve_fixpoint () =
-  print_endline ("  all cross references were solved.")
+let achieve_fixpoint unresolved_crossrefs =
+  if unresolved_crossrefs = [] then
+    print_endline ("  all cross references were solved.")
+  else
+    print_endline ("  some cross references were not solved.")
```
Comment (sakas--) [inline src/frontend/logging.ml]: Why don't you print `unresolved_crossrefs`?
I think it's informative.

Comment (qnighy) [reply src/frontend/logging.ml]: I just threw away problems like formatting and sanitization. Let's implement it simply.

**Comment by qnighy** [comment]
@gfngfn Thank you for reviewing! I implemented `probe-cross-reference` that does not record crossref resolution failure. Yet I'm not really sure the concrete situation where we need `probe-cross-reference`. Branching over crossref resolution status seems weird.

For logging part, in my opinion, logging internal things is better than logging nothing. There's still a room for improvement by allowing custom missing-cross-reference messages.

**Comment by sakas--** [comment]
I suggested logging keys because I thought it's better than logging nothing.
In my opinion, reporting elaborated messages or adding a verbose mode can be addressed by another PR.
Maybe, we should prepare a place to discuss the overall design of error reporting/logging since providing "nice error messages" should be a strong point of SATySFi.

**Comment by qnighy** [comment]
Any news on this? I'm ok with or without `probe-cross-reference`.

**Pull request merged**